test/e2e/basic_test.go

```diff
@@ -0,0 +1,71 @@
+/*
+Copyright 2018 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"log"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/GoogleCloudPlatform/spark-on-k8s-operator/pkg/apis/sparkoperator.k8s.io/v1alpha1"
+	appFramework "github.com/GoogleCloudPlatform/spark-on-k8s-operator/test/e2e/framework"
+	"github.com/stretchr/testify/assert"
+
+	"k8s.io/api/core/v1"
+)
+
+func getJobStatus(t *testing.T) v1alpha1.ApplicationStateType {
+	app, err := appFramework.GetSparkApplication(framework.SparkApplicationClient, "default", "spark-pi")
+	assert.Equal(t, nil, err)
+	return app.Status.AppState.State
+}
+
+func TestSubmitSparkPiYaml(t *testing.T) {
+	t.Parallel()
+
+	// Wait for test job to finish. Time out after 90 seconds.
+	timeout := 90
+
+	sa, err := appFramework.MakeSparkApplicationFromYaml("../../examples/spark-pi.yaml")
+	assert.Equal(t, nil, err)
+	err = appFramework.CreateSparkApplication(framework.SparkApplicationClient, "default", sa)
+	assert.Equal(t, nil, err)
+
+	status := getJobStatus(t)
+
+	timePassed := 0
+	// Update job status every 5 seconds until job is done or timeout threshold is reached.
+	for status != "COMPLETED" && timePassed <= timeout {
```

Comment:
You can use `k8s.io/apimachinery/pkg/util/wait`, which has a function named `Poll` that does exactly this kind of things.

Reply:
Thanks for the pointer!